From 86d3a723568198199ead97fa673c4ce3780734e6 Mon Sep 17 00:00:00 2001
From: Arthur Sengileyev <arthur.sengileyev@gmail.com>
Date: Tue, 21 Mar 2023 10:25:41 +0200
Subject: [PATCH] Enable QEMU machine on Windows

Signed-off-by: Arthur Sengileyev <arthur.sengileyev@gmail.com>
---
 cmd/podman/machine/platform_windows.go | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/cmd/podman/machine/platform_windows.go b/cmd/podman/machine/platform_windows.go
index 02768feb2..9fbdc2a67 100644
--- a/cmd/podman/machine/platform_windows.go
+++ b/cmd/podman/machine/platform_windows.go
@@ -7,6 +7,7 @@ import (
 	"github.com/containers/common/pkg/config"
 	"github.com/containers/podman/v4/pkg/machine"
 	"github.com/containers/podman/v4/pkg/machine/hyperv"
+	"github.com/containers/podman/v4/pkg/machine/qemu"
 	"github.com/containers/podman/v4/pkg/machine/wsl"
 )
 
@@ -34,6 +35,8 @@ func GetSystemProvider(vmType *machine.VMType) (machine.VirtProvider, error) {
 		return wsl.GetWSLProvider(), nil
 	case machine.HyperVVirt:
 		return hyperv.GetVirtualizationProvider(), nil
+	case machine.QemuVirt:
+		return qemu.GetVirtualizationProvider(), nil
 	default:
 		return nil, fmt.Errorf("unsupported virtualization provider: `%s`", resolvedVMType.String())
 	}
-- 
2.40.0

