From 771db2623bffa7a84c981208c0a7b4f681886ad3 Mon Sep 17 00:00:00 2001
From: Arthur Sengileyev <arthur.sengileyev@gmail.com>
Date: Sat, 1 Apr 2023 23:49:13 +0300
Subject: [PATCH 3/3] Enable QEMU Podman machine on Windows

Signed-off-by: Arthur Sengileyev <arthur.sengileyev@gmail.com>
---
 cmd/podman/machine/platform_windows.go | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/cmd/podman/machine/platform_windows.go b/cmd/podman/machine/platform_windows.go
index a50f0efd0..a47b5c4e7 100644
--- a/cmd/podman/machine/platform_windows.go
+++ b/cmd/podman/machine/platform_windows.go
@@ -7,6 +7,7 @@ import (
 	"github.com/containers/common/pkg/config"
 	"github.com/containers/podman/v4/pkg/machine"
 	"github.com/containers/podman/v4/pkg/machine/hyperv"
+	"github.com/containers/podman/v4/pkg/machine/qemu"
 	"github.com/containers/podman/v4/pkg/machine/wsl"
 )
 
@@ -29,6 +30,8 @@ func GetSystemProvider() (machine.VirtProvider, error) {
 		return wsl.GetWSLProvider(), nil
 	case machine.HyperVVirt:
 		return hyperv.GetVirtualizationProvider(), nil
+	case machine.QemuVirt:
+		return qemu.GetVirtualizationProvider(), nil
 	default:
 		return nil, fmt.Errorf("unsupported virtualization provider: `%s`", resolvedVMType.String())
 	}
-- 
2.40.0

