From 2e6d647855d2a2a6be39837507984e90fd0b6728 Mon Sep 17 00:00:00 2001
From: Arthur Sengileyev <arthur.sengileyev@gmail.com>
Date: Mon, 8 Jul 2024 18:52:50 +0300
Subject: [PATCH 3/3] Disable FS mounts by default in tests (hack)

Signed-off-by: Arthur Sengileyev <arthur.sengileyev@gmail.com>
---
 pkg/machine/e2e/config_test.go | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/pkg/machine/e2e/config_test.go b/pkg/machine/e2e/config_test.go
index 3172601c5..b15c6926b 100644
--- a/pkg/machine/e2e/config_test.go
+++ b/pkg/machine/e2e/config_test.go
@@ -198,6 +198,10 @@ func (m *machineTestBuilder) run() (*machineSession, error) {
 }
 
 func runWrapper(podmanBinary string, cmdArgs []string, timeout time.Duration, wait bool) (*machineSession, error) {
+	if len(cmdArgs) >= 2 && cmdArgs[0] == "machine" && cmdArgs[1] == "init" && !slices.Contains(cmdArgs, "-v") {
+		tailArgs := append([]string{"-v", ""}, cmdArgs[2:]...)
+		cmdArgs = append([]string{"machine", "init"}, tailArgs...)
+	}
 	if len(os.Getenv("DEBUG")) > 0 {
 		cmdArgs = append([]string{"--log-level=debug"}, cmdArgs...)
 	}
-- 
2.45.2

