From 9a452db91804e9a8fd57e89db4adeb8a5237a834 Mon Sep 17 00:00:00 2001
From: Mario Loriedo <mario.loriedo@gmail.com>
Date: Wed, 21 May 2025 11:10:59 +0200
Subject: [PATCH 1/3] Support '$FOCUS' env variable on winmake too

Signed-off-by: Mario Loriedo <mario.loriedo@gmail.com>
---
 winmake.ps1 | 9 +++++++--
 1 file changed, 7 insertions(+), 2 deletions(-)

diff --git a/winmake.ps1 b/winmake.ps1
index 4ab6ae2afd..a02fe344e3 100644
--- a/winmake.ps1
+++ b/winmake.ps1
@@ -78,10 +78,15 @@ function Local-Machine {
     );
     Build-Ginkgo
     if ($files) {
-         $files = " --focus-file $files "
+         $files = "--focus-file ""$files"""
+    } elseif ($FOCUS_FILE) {
+        $files = "--focus-file ""$FOCUS_FILE"" --silence-skips"
+    }
+    if ($FOCUS) {
+        $focus = "--focus ""$FOCUS"" --silence-skips"
     }
 
-    Run-Command "./bin/ginkgo.exe -vv  --tags `"$remotetags`" --timeout=90m --trace --no-color $files pkg/machine/e2e/."
+    Run-Command "./bin/ginkgo.exe -vv  --tags `"$remotetags`" --timeout=90m --trace --no-color $focus $files pkg/machine/e2e/."
 }
 
 # Expect starting directory to be /podman

From af495e1aa9650c10b9189d14e931e82e07ac9ea8 Mon Sep 17 00:00:00 2001
From: Mario Loriedo <mario.loriedo@gmail.com>
Date: Wed, 21 May 2025 11:12:14 +0200
Subject: [PATCH 2/3] Replace alpine_nginx with TESTIMAGE in e2e tests

The image quay.io/libpod/alpine_nginx doesn't have a `linux/arm64` version
and the test "Single character volume mount" on Windows ARM64 was
failing. Changing it to TESTIMAGE (quay.io/libpod/testimage) fixes it.

Signed-off-by: Mario Loriedo <mario.loriedo@gmail.com>
---
 pkg/machine/e2e/basic_test.go | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/pkg/machine/e2e/basic_test.go b/pkg/machine/e2e/basic_test.go
index f3113f64d3..408096ed91 100644
--- a/pkg/machine/e2e/basic_test.go
+++ b/pkg/machine/e2e/basic_test.go
@@ -113,7 +113,7 @@ var _ = Describe("run basic podman commands", func() {
 		Expect(err).ToNot(HaveOccurred())
 		Expect(volumeCreate).To(Exit(0))
 
-		run, err := mb.setCmd(bm.withPodmanCommand([]string{"run", "-v", "a:/test:Z", "quay.io/libpod/alpine_nginx", "true"})).run()
+		run, err := mb.setCmd(bm.withPodmanCommand([]string{"run", "-v", "a:/test:Z", TESTIMAGE, "true"})).run()
 		Expect(err).ToNot(HaveOccurred())
 		Expect(run).To(Exit(0))
 	})

From 9872f0e83c0e833c322c7f956dafddebe78a1f9d Mon Sep 17 00:00:00 2001
From: Mario Loriedo <mario.loriedo@gmail.com>
Date: Wed, 21 May 2025 14:36:38 +0200
Subject: [PATCH 3/3] Update expected output for a machine copy test

The podman machine copy test "attempt copying file to a new directory"
was failing because on recent version of Windows the error message
doesn't match the expected error message. To make it work on new and
old version of Windows both old and new error messages are now
considered as valid.

Fixes https://github.com/containers/podman/issues/26056

Signed-off-by: Mario Loriedo <mario.loriedo@gmail.com>
---
 pkg/machine/e2e/cp_test.go | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/pkg/machine/e2e/cp_test.go b/pkg/machine/e2e/cp_test.go
index 2aa8c2b193..808e96bdcf 100644
--- a/pkg/machine/e2e/cp_test.go
+++ b/pkg/machine/e2e/cp_test.go
@@ -357,7 +357,8 @@ var _ = Describe("run cp commands", func() {
 		switch runtime.GOOS {
 		case "windows":
 			hostDirPath = filepath.ToSlash(hostDirPath)
-			fallthrough
+			Expect(session.errorToString()).To(Or(ContainSubstring(fmt.Sprintf("scp: open local \"%s\": No such file or directory", hostDirPath)),
+				ContainSubstring(fmt.Sprintf("scp: open local \"%s\": Unknown error", hostDirPath))))
 		case "darwin":
 			Expect(session.errorToString()).To(ContainSubstring(fmt.Sprintf("scp: open local \"%s\": No such file or directory", hostDirPath)))
 		case "linux":
