From c09d3a1f855c5a6621d290c3f0c365c5b0f0de40 Mon Sep 17 00:00:00 2001
From: Arthur Sengileyev <arthur.sengileyev@gmail.com>
Date: Thu, 4 Dec 2025 16:37:40 +0200
Subject: [PATCH] Remove `ftruncate` macro

---
 include/system/os-win32.h | 4 ----
 1 file changed, 4 deletions(-)

diff --git a/include/system/os-win32.h b/include/system/os-win32.h
index 22d72babdf..396d4d28ea 100644
--- a/include/system/os-win32.h
+++ b/include/system/os-win32.h
@@ -140,10 +140,6 @@ static inline void os_setup_limits(void)
 
 int qemu_ftruncate64(int, int64_t);
 
-#if !defined(ftruncate)
-# define ftruncate qemu_ftruncate64
-#endif
-
 static inline char *realpath(const char *path, char *resolved_path)
 {
     _fullpath(resolved_path, path, _MAX_PATH);
-- 
2.52.0

