From 0b1b3d4f2ea052ad10090d76b276d7c887908876 Mon Sep 17 00:00:00 2001
From: Arthur Sengileyev <arthur.sengileyev@gmail.com>
Date: Wed, 17 Aug 2022 17:49:21 +0300
Subject: [PATCH] Allow unix sockets on windows

Signed-off-by: Arthur Sengileyev <arthur.sengileyev@gmail.com>
---
 pkg/transport/listen_windows.go | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/pkg/transport/listen_windows.go b/pkg/transport/listen_windows.go
index 30ee11e..423f887 100644
--- a/pkg/transport/listen_windows.go
+++ b/pkg/transport/listen_windows.go
@@ -25,6 +25,8 @@ func Listen(endpoint string) (net.Listener, error) {
 			VMID:      hvsock.GUIDWildcard,
 			ServiceID: svcid,
 		})
+	case "unix":
+		return net.Listen(parsed.Scheme, parsed.Path)
 	case "tcp":
 		return net.Listen("tcp", parsed.Host)
 	default:
-- 
2.37.2

