From bdf02a0eeead218d2b3fcee5cfa9f7f2fa77b1d8 Mon Sep 17 00:00:00 2001
From: Arthur Sengileyev <arthur.sengileyev@gmail.com>
Date: Tue, 11 Feb 2025 22:46:40 +0200
Subject: [PATCH 2/2] Add quickfix for Windows compat `limactl stop`

---
 pkg/qemu/qemu.go | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/pkg/qemu/qemu.go b/pkg/qemu/qemu.go
index 1d8fea6a..ed4390cf 100644
--- a/pkg/qemu/qemu.go
+++ b/pkg/qemu/qemu.go
@@ -1080,7 +1080,7 @@ func qemuEdk2Arch(arch limayaml.Arch) string {
 }
 
 func Exe(arch limayaml.Arch) (exe string, args []string, err error) {
-	exeBase := "qemu-system-" + qemuArch(arch)
+	exeBase := "qemu-system-" + qemuArch(arch) + "w"
 	envK := "QEMU_SYSTEM_" + strings.ToUpper(qemuArch(arch))
 	if envV := os.Getenv(envK); envV != "" {
 		ss, err := shellwords.Parse(envV)
-- 
2.48.1

