From b05bb1a70b1dcf2c82cf91da4b88bc19ec6693aa Mon Sep 17 00:00:00 2001
From: Arthur Sengileyev <arthur.sengileyev@gmail.com>
Date: Sat, 8 Mar 2025 23:29:38 +0200
Subject: [PATCH 3/4] Add additional PATH env var for tooling on Windows

Signed-off-by: Arthur Sengileyev <arthur.sengileyev@gmail.com>
---
 cmd/limactl/main.go | 10 ++++++++++
 1 file changed, 10 insertions(+)

diff --git a/cmd/limactl/main.go b/cmd/limactl/main.go
index 268fad0d..576489c3 100644
--- a/cmd/limactl/main.go
+++ b/cmd/limactl/main.go
@@ -28,6 +28,16 @@ const (
 )
 
 func main() {
+	if runtime.GOOS == "windows" {
+		extras, hasExtra := os.LookupEnv("_LIMA_WINDOWS_EXTRA_PATH")
+		if hasExtra && strings.TrimSpace(extras) != "" {
+			p := os.Getenv("PATH")
+			err := os.Setenv("PATH", strings.TrimSpace(extras)+string(filepath.ListSeparator)+p)
+			if err != nil {
+				logrus.Warning("Can't add extras to PATH, relying fully on system PATH")
+			}
+		}
+	}
 	if err := newApp().Execute(); err != nil {
 		handleExitCoder(err)
 		logrus.Fatal(err)
-- 
2.48.1

