From b236b72f31a0e5633c60eec7cabfc7453d7ae6ab Mon Sep 17 00:00:00 2001
From: Arthur Sengileyev <arthur.sengileyev@gmail.com>
Date: Sun, 23 Mar 2025 21:58:16 +0200
Subject: [PATCH 1/2] Disable ControlPersist for sshfs connections on Windows

Signed-off-by: Arthur Sengileyev <arthur.sengileyev@gmail.com>
---
 pkg/hostagent/mount.go | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/pkg/hostagent/mount.go b/pkg/hostagent/mount.go
index cfc3f3bb..0ad6e77d 100644
--- a/pkg/hostagent/mount.go
+++ b/pkg/hostagent/mount.go
@@ -68,6 +68,9 @@ func (a *HostAgent) setupMount(m limayaml.Mount) (*mount, error) {
 		Readonly:            !(*m.Writable),
 		SSHFSAdditionalArgs: []string{"-o", sshfsOptions},
 	}
+	if runtime.GOOS == "windows" {
+		rsf.SSHConfig.Persist = false
+	}
 	if err := rsf.Prepare(); err != nil {
 		return nil, fmt.Errorf("failed to prepare reverse sshfs for %q on %q: %w", resolvedLocation, *m.MountPoint, err)
 	}
-- 
2.49.0

