From ffb66c42509c3575d74ca78e562e9085f1ecbd5e Mon Sep 17 00:00:00 2001
From: Arthur Sengileyev <arthur.sengileyev@gmail.com>
Date: Sun, 16 Feb 2025 22:14:10 +0200
Subject: [PATCH] Ignore errors during limactl stop

---
 hack/test-templates.sh | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/hack/test-templates.sh b/hack/test-templates.sh
index 6a08c0e9..3c90db88 100755
--- a/hack/test-templates.sh
+++ b/hack/test-templates.sh
@@ -377,7 +377,7 @@ if [[ -n ${CHECKS["restart"]} ]]; then
 	fi
 
 	INFO "Stopping \"$NAME\""
-	limactl stop "$NAME"
+	limactl stop "$NAME" || true
 	sleep 3
 
 	if [[ -n ${CHECKS["disk"]} ]]; then
@@ -491,7 +491,7 @@ if [[ $NAME == "fedora" && "$(limactl ls --json "$NAME" | jq -r .vmType)" == "vz
 fi
 
 INFO "Stopping \"$NAME\""
-limactl stop "$NAME"
+limactl stop "$NAME" || true
 sleep 3
 
 INFO "Deleting \"$NAME\""
-- 
2.48.1

