From 0e7050648bc9bb24b6f2a9ffb20faff0ffc32913 Mon Sep 17 00:00:00 2001
From: Arthur Sengileyev <arthur.sengileyev@gmail.com>
Date: Sun, 23 Mar 2025 22:01:17 +0200
Subject: [PATCH 2/2] Enable ControlPersist for WSL2 tooling

---
 pkg/hostagent/mount.go | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/pkg/hostagent/mount.go b/pkg/hostagent/mount.go
index 0ad6e77d..2ee470ca 100644
--- a/pkg/hostagent/mount.go
+++ b/pkg/hostagent/mount.go
@@ -69,7 +69,10 @@ func (a *HostAgent) setupMount(m limayaml.Mount) (*mount, error) {
 		SSHFSAdditionalArgs: []string{"-o", sshfsOptions},
 	}
 	if runtime.GOOS == "windows" {
-		rsf.SSHConfig.Persist = false
+		localFS, err := ioutilx.IsLocalWindowsFS()
+		if err != nil || localFS {
+			rsf.SSHConfig.Persist = false
+		}
 	}
 	if err := rsf.Prepare(); err != nil {
 		return nil, fmt.Errorf("failed to prepare reverse sshfs for %q on %q: %w", resolvedLocation, *m.MountPoint, err)
-- 
2.49.0

