From 3185091ed5e0dad94746d35d72418e477da3ba6e Mon Sep 17 00:00:00 2001
From: Arthur Sengileyev <arthur.sengileyev@gmail.com>
Date: Sun, 23 Mar 2025 22:44:13 +0200
Subject: [PATCH] Unify Lima user home inside guest for all platforms

Signed-off-by: Arthur Sengileyev <arthur.sengileyev@gmail.com>
---
 pkg/osutil/user.go | 40 ++--------------------------------------
 1 file changed, 2 insertions(+), 38 deletions(-)

diff --git a/pkg/osutil/user.go b/pkg/osutil/user.go
index 8a47ffee..ca084f9e 100644
--- a/pkg/osutil/user.go
+++ b/pkg/osutil/user.go
@@ -7,15 +7,12 @@ import (
 	"fmt"
 	"os/exec"
 	"os/user"
-	"path"
-	"path/filepath"
 	"regexp"
 	"runtime"
 	"strconv"
 	"strings"
 	"sync"
 
-	"github.com/lima-vm/lima/pkg/ioutilx"
 	. "github.com/lima-vm/lima/pkg/must"
 	"github.com/lima-vm/lima/pkg/version/versionutil"
 	"github.com/sirupsen/logrus"
@@ -115,9 +112,8 @@ func LimaUser(limaVersion string, warn bool) *user.User {
 			warnings = append(warnings, warning)
 			limaUser.Username = fallbackUser
 		}
-		if runtime.GOOS != "windows" {
-			limaUser.HomeDir = "/home/{{.User}}.linux"
-		} else {
+		limaUser.HomeDir = "/home/{{.User}}.linux"
+		if runtime.GOOS == "windows" {
 			idu, err := call([]string{"id", "-u"})
 			if err != nil {
 				logrus.Debug(err)
@@ -146,38 +142,6 @@ func LimaUser(limaVersion string, warn bool) *user.User {
 				warnings = append(warnings, warning)
 				limaUser.Gid = formatUidGid(gid)
 			}
-			home, err := ioutilx.WindowsSubsystemPath(limaUser.HomeDir)
-			if err != nil {
-				logrus.Debug(err)
-			} else {
-				// Trim mount prefix within Subsystem
-				// cygwin/msys2 cygpath could have prefix for drive mounts configured via /etc/fstab
-				// wsl wslpath could have prefix for drive mounts configured via [automount] section in wsl.conf
-				drivePath, err := ioutilx.WindowsSubsystemPath(filepath.VolumeName(limaUser.HomeDir) + "/")
-				if err != nil {
-					logrus.Debug(err)
-				} else {
-					prefix := path.Dir(strings.TrimSuffix(drivePath, "/"))
-					if prefix != "/" {
-						home = strings.TrimPrefix(home, prefix)
-					}
-					home += ".linux"
-				}
-			}
-			if home == "" {
-				drive := filepath.VolumeName(limaUser.HomeDir)
-				home = filepath.ToSlash(limaUser.HomeDir)
-				// replace C: with /c
-				prefix := strings.ToLower(fmt.Sprintf("/%c", drive[0]))
-				home = strings.Replace(home, drive, prefix, 1)
-				home += ".linux"
-			}
-			if !regexPath.MatchString(limaUser.HomeDir) {
-				warning := fmt.Sprintf("local home %q is not a valid Linux path (must match %q); using %q home instead",
-					limaUser.HomeDir, regexPath.String(), home)
-				warnings = append(warnings, warning)
-				limaUser.HomeDir = home
-			}
 		}
 	})
 	if warn {
-- 
2.49.0

